### DISCUSSION TITLE
Subscription Extension

### DISCUSSION THREAD

**Discussion by GWLdeveloperAjeetesh** [kinngh/shopify-node-express-mongodb-app#103] [General]
Hi @kinngh I want to add product subscription Extension. I tried to add CLI but it is throwing error, what could be the alternate solution for that?How can i do that? I explored shopify doc but it seems very confusing and they are suggesting to go with CLI so what should i do?

**Comment by kinngh** [comment]
What is the error? You need to add screenshots, error logs, etc before anyone can help you. You have the context, not me^

Also, this should be an issue, not a discussion. Especially not under ideas.

**Comment by GWLdeveloperAjeetesh** [comment]
Might be, but i haven't saw anything related to the product subscription extension in the boilerplate, so i thought this is something new.
And for now i am following this doc:
https://shopify.dev/docs/apps/selling-strategies/purchase-options/app-extensions/getting-started
where they are asking me to give command npm run dev, but the issue is that the port is already occupied, Is it good to add two port at the same time in the same app?
![Uploading image.png…]()

**Reply by kinngh** [reply]
You need to change the dir structure of your entire app, the same way Shopify suggests. Make a new parent folder and throw your express app in there. Your extensions go in a separate structure.

`my_main_app/`
- `/express app/`
- `/extensions/`

In your `extensions/`, run `npm init --y` and then run `shopify app generate extension`. Port over your extension folder from inside of your express app into here, and then run `shopify app dev` in the `extensions/` folder for it to create a new cloudflare tunnel instance that you can now use to run in dev mode, while your express app runs completely separately.
